### PR TITLE
Add `tolerance` argument to `*_if_equal()` functions

### LINKED ISSUE #294: `pass_if_equal()` prone to floating point errors
Functions that call `grade_if_equal()` can have unexpected results when code is subject to floating point errors:

``` r
library(gradethis)

grade_this({
  pass_if_equal()
  fail()
})(
  mock_this_exercise(.user_code = "sqrt(2) ^ 2", .solution_code = "2")
)
#> <gradethis_graded: [Incorrect]
#>   Incorrect. I didn't expect `^` where you wrote `sqrt(2)^2`. Try it
#>   again; next time's the charm!
#> >
```

<sup>Created on 2022-03-09 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

This seems to be because [`grade_if_equal()` calls `waldo::compare()` without setting the `tolerance` argument](https://github.com/rstudio/gradethis/blob/fea404be2c5bd6d14366108838e9e469eb608fea/R/graded.R#L396). This is akin to using `identical()` rather than `all.equal()`.

Is this intended behavior or a bug? Should we add a `tolerance` argument to the `_if_equal()` functions?

### PR DESCRIPTION
**Pull request by rossellhayes**
``` r
library(gradethis)

pass_if_equal(y = 2, x = sqrt(2) ^ 2)
#> <gradethis_graded: [Correct] Superb work! Correct!>

pass_if_equal(y = 2, x = sqrt(2) ^ 2, tolerance = NULL)
#> NULL
```

<sup>Created on 2022-03-10 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Closes #294.

### CODE REVIEW

File: R/graded.R [hunk 1]
```diff
@@ -360,7 +362,10 @@ pass_if_equal <- function(
     assert_object_found_in_env(y, env, "pass_if_equal")
   }
   maybe_extras(
-    grade_if_equal(x = x, y = y, message = message, correct = TRUE, env = env, ...),
+    grade_if_equal(
+      x = x, y = y,
+      message = message, correct = TRUE, env = env, tolerance = tolerance, ...
+    ),
```
Comment (gadenbuie) [inline R/graded.R:368]: Style nit: arguments all on the same line or one line each

```suggestion
    grade_if_equal(
      x = x, 
      y = y,
      message = message,
      correct = TRUE,
      env = env,
      tolerance = tolerance, 
      ...
    ),
```

```suggestion
    grade_if_equal(x = x, y = y, message = message, correct = TRUE, env = env, tolerance = tolerance, ...),
```

Comment (gadenbuie) [reply R/graded.R:368]: (same for  `fail_if_equal()` and `grade_if_equal()`)

**Review by gadenbuie — APPROVED** [review]
Looks great, just a small style nit

File: R/graded.R (hunk 1 above)
Comment (rossellhayes) [reply R/graded.R]: Updated in [cc12cf8](https://github.com/rstudio/gradethis/pull/295/commits/cc12cf8653c78518e9444a68ee07f4e31db7eefd)

**Pull request merged**